frontend/packages/ux-editor/src/components/Elements/SubForm/CreateSubFormWrapper.tsx

```diff
@@ -0,0 +1,77 @@
+import React, { useState } from 'react';
+import { StudioButton, StudioPopover, StudioTextfield } from '@studio/components';
+import { PlusIcon } from '@studio/icons';
+import { useAddLayoutSetMutation } from 'app-development/hooks/mutations/useAddLayoutSetMutation';
+import { useStudioEnvironmentParams } from 'app-shared/hooks/useStudioEnvironmentParams';
+import { useTranslation } from 'react-i18next';
+import { useValidateLayoutSetName } from 'app-shared/hooks/useValidateLayoutSetName';
+import type { LayoutSets } from 'app-shared/types/api/LayoutSetsResponse';
+import classes from './CreateSubFormWrapper.module.css';
+
+type CreateSubFormWrapperProps = {
+  layoutSets: LayoutSets;
+  onSubFormCreated: (layoutSetName: string) => void;
+};
+
+export const CreateSubFormWrapper = ({
+  layoutSets,
+  onSubFormCreated,
+}: CreateSubFormWrapperProps) => {
+  const [createNewOpen, setCreateNewOpen] = useState(false);
+  const [newSubFormName, setNewSubFormName] = useState('');
+  const [nameError, setNameError] = useState('');
+  const { t } = useTranslation();
+  const { validateLayoutSetName } = useValidateLayoutSetName();
+  const { org, app } = useStudioEnvironmentParams();
+  const { mutate: addLayoutSet } = useAddLayoutSetMutation(org, app);
+
+  const onCreateConfirmClick = () => {
+    setCreateNewOpen(false);
+
+    addLayoutSet({
+      layoutSetIdToUpdate: newSubFormName,
+      layoutSetConfig: {
+        id: newSubFormName,
+        type: 'subform',
+      },
+    });
+    onSubFormCreated(newSubFormName);
+  };
+
+  const onNameChange = (subFormName: string) => {
+    const subFormNameValidation = validateLayoutSetName(subFormName, layoutSets);
+    setNameError(subFormNameValidation);
+    setNewSubFormName(subFormName);
+  };
+
+  return (
+    <StudioPopover open={createNewOpen} onOpenChange={setCreateNewOpen}>
```

Comment:
I believe we agreed on using modal and not popovers? 🧐

Reply:
My point of view, this is OK for now, since we are going to remove this code before production anyway. But this PR enables us to continue working on blocked issues. This feature is behind feature flag. 

It's more value to merge this PR with StudioPopover than rewrite it to modal. :)